Bukkit/src/main/java/com/plotsquared/bukkit/listener/PlayerEventListener.java

```diff
@@ -204,6 +206,29 @@ public PlayerEventListener(
         this.plotListener = plotListener;
     }
 
+    @EventHandler(ignoreCancelled = true)
+    public void onEffect(@NonNull EntityPotionEffectEvent event) {
+        if (Settings.Enabled_Components.DISABLE_BEACON_EFFECT_OVERFLOW ||
+                event.getCause() != EntityPotionEffectEvent.Cause.BEACON ||
+                !(event.getEntity() instanceof Player player)) {
+            return;
+        }
+
+        UUID uuid = player.getUniqueId();
+        PotionEffect effect = event.getNewEffect();
+        if (effect == null) {
+            PotionEffect oldEffect = event.getOldEffect();
+            if (oldEffect != null) {
+                String name = oldEffect.getType().getName();
+                this.addEffect(uuid, name, -1);
```

Comment:
How did you test this out? #addEffect isn't accessible in the Bukkit module the way you implemented it in the Core module.